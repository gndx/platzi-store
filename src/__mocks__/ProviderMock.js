import React from 'react';
import { createStore } from 'redux';

import { Router } from 'react-router-dom';

import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import initialState from '../initialState';
import reducer from '../reducers';

const store = createStore(reducer, initialState);
const history = createBrowserHistory();
const ProviderMock =props  => (
  <Provider store={store}>
    <Router history={history}>
        {props.children}
    </Router>
  </Provider>
);

export default ProviderMock