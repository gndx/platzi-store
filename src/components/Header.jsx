import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import '../styles/components/Header.styl';

const Header = props => (
  <div className="Header">
    <h1 className="Header-title">
      <Link to="/">
        Platzi Store
      </Link>
    </h1>
    <div className="Header-checkout">
      <Link to="/checkout">
        <i className="fas fa-shopping-basket" />
      </Link>
      {props.cart.length > 0 &&
        <div className="Header-alert">{props.cart.length}</div>
      }
    </div>
  </div>
);

const mapStateToProps = (state) => {
  return {
    cart: state.cart,
  };
};

export default connect(mapStateToProps, null)(Header);
