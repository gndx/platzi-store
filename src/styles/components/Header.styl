.Header
  display flex
  justify-content space-between
  &-checkout
    display flex
    justify-content space-evenly
    align-items center
    justify-items center
    width 30px
    margin 0 0 0 .5em
  &-alert
    color #33b13a
    font-size 14px
    margin-left 5px
  a
    text-decoration none
    color #33b1ff

