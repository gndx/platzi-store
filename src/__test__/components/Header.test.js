import React from 'react';
import { mount, shallow } from 'enzyme';
import ProviderMock from '../../__mocks__/ProviderMock';
import { create } from 'react-test-renderer';
import Header from '../../components/Header';

describe('<Header />', () => {
    test('Render del componente Header', () => {
        const header = shallow(
            <ProviderMock>
                <Header />
            </ProviderMock>,
        );
        expect(header.length).toEqual(1);
    });
    test('Render del Titulo', () => {
        const header = mount(
            <ProviderMock>
                <Header />
            </ProviderMock>
        );
        expect(header.find(".Header-title").text()).toEqual("Platzi Store");
    });
});

describe('Header Snapshot', () => {
    test('Comprobar el Snapshot de Header', () => {
        const header = create(
            <ProviderMock>
                <Header />
            </ProviderMock>
        );
        expect(header.toJSON()).toMatchSnapshot();
    });
});