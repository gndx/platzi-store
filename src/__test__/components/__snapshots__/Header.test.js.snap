// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot Comprobar el Snapshot de Header 1`] = `
<div
  className="Header"
>
  <h1
    className="Header-title"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Platzi Store
    </a>
  </h1>
  <div
    className="Header-checkout"
  >
    <a
      href="/checkout"
      onClick={[Function]}
    >
      <i
        className="fas fa-shopping-basket"
      />
    </a>
  </div>
</div>
`;
